### ISSUE TITLE
[FEATURE] Support swing.mh.remains and swing.oh.remains

### ISSUE THREAD

**Issue by martinioo** [Hekili/hekili#1249]
### Before You Begin

- [X] I confirm that I have downloaded the latest version of the addon.
- [X] I checked for an [existing, open ticket](https://github.com/Hekili/hekili/labels/enhancement) for this request and was not able to find one.
- [X] I edited the title of this feature request (above) so that it describes the issue I am reporting.

### Feature Request

**ARMS Warrior Charge Weaving to generate more Rage**

I would really appriciate an Implimatation of this function:

Ref. Method ARMS Warrior Guide:
Charge Weaving:
- Wait for an auto-attack (Swing-Timer!) to complete before running away from the boss to
- Charge back in.
- Keep yourself at max melee range of your target as much as possible. This will reduce the amount you need to run and will make it less likely your next auto attack is late.

Other Source:
Charge weave when mortal strike on cooldown, less than 2x overpower charges, and auto attack just went off

Exampe Code (not working / Ref. Tier19 Arms Warrior):
`- actions.single_target+=/charge,if=rage.deficit>=50&swing.mh.remains>1.5&cooldown.mortal_strike.remains>=gcd&overpower.stack<2`

**(or overpower.charges<2??)**

Thx!!!!

### Additional Information

Discussed in Discord

### Contact Information

martinioo#8753

**Comment by Hekili** [comment]
Just wanted to confirm I've looked at this a little bit.  Swing timers (and downstream impacts of swing timers) are clumsily implemented in the addon, tbh.  This might be something I'll address in the near future since I just had to resolve a separate bug related to offhand swings for specs not wielding an offhand.  I'll post if/when anything changes.

One issue that we will probably see is that let's say you have a 2.6s swing time, and there's 1.3s until your next swing.  What happens in 1.3s?

1. Your swing timer is 0; the swing is happening now.
2. Your swing timer is 2.6; the swing just happened and the timer reset.

If it's 1, when does the swing timer reset?  If it's 2, how do I check that a swing is happening right now?

Both of these have implications for the profile.  Checking a condition like the one you're showing, `swing.mh.remains > 1.5` would not really be a problem since that effectively means "we're in the 1.1s window after a swing took place" in my 2.6s window.

I'm not sure what the simc implementation does or did.  I'll take a look later.
